pureconfig/src/main/scala/com/avast/clients/rabbitmq/pureconfig/PureconfigImplicits.scala

```diff
@@ -0,0 +1,94 @@
+package com.avast.clients.rabbitmq.pureconfig
+
+import _root_.pureconfig.ConfigReader.Result
+import _root_.pureconfig._
+import _root_.pureconfig.generic.ProductHint
+import _root_.pureconfig.generic.semiauto._
+import com.avast.clients.rabbitmq.api.DeliveryResult
+import com.avast.clients.rabbitmq.api.DeliveryResult._
+import com.avast.clients.rabbitmq.{
+  AutoBindExchangeConfig,
+  AutoBindQueueConfig,
+  AutoDeclareExchangeConfig,
+  AutoDeclareQueueConfig,
+  BindArgumentsConfig,
+  BindExchangeConfig,
+  BindQueueConfig,
+  ConsumerConfig,
+  CredentialsConfig,
+  DeclareArgumentsConfig,
+  DeclareExchangeConfig,
+  DeclareQueueConfig,
+  ExchangeType,
+  NetworkRecoveryConfig,
+  ProducerConfig,
+  ProducerPropertiesConfig,
+  PullConsumerConfig,
+  RabbitMQConnectionConfig,
+  RecoveryDelayHandlers,
+  SslConfig,
+  TrustStoreConfig
+}
+import com.rabbitmq.client.RecoveryDelayHandler
+import org.slf4j.event.Level
+
+private[pureconfig] class PureconfigImplicits(implicit namingConvention: NamingConvention = CamelCase) {
```

Comment:
also, why do we even need this at all? PureConfig loading should work automatically (at least for vast majority of use cases) without any need for manual/semiautomatic implicits

Reply:
We have decided in the past that fully automatic mode is not going to be used. There are reasons of course. One of them is speed of compilation because the fully automatic mode sometimes means that readers are derived again and again instead of one being derived in the `val`.  Another one is that this allows you to have at least some control over the readers. For example you might want to have a special reader for `String` (e.g. hex-string) so you could put it here and be sure that it will work like this for your library, in fully automatic mode this is not possible and it depends on your imports whether `String` is read as string or hex-string. The few lines of code you need to write seem to be on the safer side even though it's less ergonomic to do.